EIPS/eip-4750.md

```diff
@@ -29,12 +29,12 @@ Furthermore it aims to improve analysis opportunities by encoding the number of
 
 ### EOF container changes
 
-1. The requirement of [EIP-3540](./eip-3540.md) "Exactly one code section MUST be present." is relaxed to "At least one code section MUST be present.", i.e. multiple code sections (`kind = 1`) are allowed.
-2. Total number of code sections MUST NOT exceed 1024.
+1. The [EIP-3540](./eip-3540.md) code section header is redefined to be an array of code section sizes terminated with a `0x00` byte. E.g. `code_section_header := 1, (code_section_size)+, 0`.
```

Comment:
Is this a compression method so that there will be still a single code section header? Why doesn't this rule apply to the data section then? And if so, why not just introduce a simple RLE compression 😅

Reply:
I believe there can only be one data section so this rule wouldn't be able to apply? This is less about compression and more about having a intuitive header format. Practically, the code sections must be sandwiched between the types and data sections. It feels quite natural to just have the code header dynamically list all the sizes. In fact, requiring the type section might not even be such a bad idea. That would simplify the decoding a bit since you can always assume the size of the type section (no longer optional) is `2n` the number of code sections.

Reply:
> In fact, requiring the type section might not even be such a bad idea. That would simplify the decoding a bit since you can always assume the size of the type section (no longer optional) is 2n the number of code sections.

It can stay optional, because omitted type section implies there's just 1 code section.

(but I think it still makes sense to make it required for explicitness and simplifying the parser)

Reply:
Right, so for that reason I think it should be required.